pydantic/main.py

```diff
@@ -652,16 +645,16 @@ def to_string(self, pretty: bool = False) -> str:
         return '{}{}{}'.format(
             self.__class__.__name__,
             divider,
-            divider.join('{}={}'.format(k, truncate(v)) for k, v in self.__values__.items()),
+            divider.join('{}={}'.format(k, truncate(v)) for k, v in self.__dict__.items()),
         )
 
     def __str__(self) -> str:
         return self.to_string()
 
-    def __dir__(self) -> 'ListStr':
-        ret = list(object.__dir__(self))
-        ret.extend(self.__values__.keys())
-        return ret
+    @property
```

Comment:
I guess we can leave this in for a couple of releases, I doubt it's necessary but no reason to remove it.

I assume with this change `vars(model)` and `dir(model)` still work correct, but have we checked that?

Reply:
We have `test_dir_fields` in `tests/test_main.py`, so yeah, I checked that is's working